File: little-bee/controller/router/agent.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_AGENT = "/config/agent"
const URL_ROUTER_AGENT_BY_ID = "/config/agent/:agent-id"
const URL_ROUTER_AGENT_QUERY = "/config/agent/query"
const URL_ROUTER_AGENT_ALL = "/config/agent/all"

func AgentRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_AGENT, config.CreateAgent)
	r.GET(URL_ROUTER_AGENT_BY_ID, config.RetrieveAgent)
	r.PUT(URL_ROUTER_AGENT, config.UpdateAgent)
	r.DELETE(URL_ROUTER_AGENT_BY_ID, config.DeleteAgent)
	r.PUT(URL_ROUTER_AGENT_BY_ID, config.DeleteAgentWithFlag)
	r.POST(URL_ROUTER_AGENT_QUERY, config.QueryAgent)
	r.GET(URL_ROUTER_AGENT_ALL, config.GetAllAgents)
	return r
}

File: frontend/console/src/api/agent.js
import Axios from "@/http";

export function createAgent(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/agent', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveAgent(agentId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/agent/${ agentId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateAgent(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/agent', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteAgent(agentId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/agent/${ agentId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteAgentWithFlag(agentId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/agent/${ agentId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllAgents() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/agent/all').then(res => {
            resolve(res)
        })
    })
}

export function queryAgent (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/agent/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}